From bbfe92d81cc313f2859eb776cc832b7cac962c3c Mon Sep 17 00:00:00 2001
From: Megha Dey <megha.dey@linux.intel.com>
Date: Tue, 19 Jun 2018 12:55:07 -0700
Subject: [PATCH] BITS: GRUB: prepare to use a memdisk

When running from a memdisk, the memdisk module is needed. Likewise, (memdisk)
should be specified as the filesystem root. These two preparations are done
when building the GRUB image. Placeholders are used so that builds can be done
for both physical disk and memdisk. Such placeholders will be edited by bitbake
at build time.

Signed-off-by: Ricardo Neri <ricardo.neri-calderon@linux.intel.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 814ee51..a840f23 100644
--- a/Makefile
+++ b/Makefile
@@ -112,7 +112,7 @@ i386-pc-img:=boot/grub/core.img
 i386-efi-img:=efi/boot/bootia32.efi
 x86_64-efi-img:=efi/boot/bootx64.efi
 i386-pc-extra-modules:=biosdisk
-common-modules:=fat part_msdos part_gpt iso9660
+common-modules:=fat part_msdos part_gpt iso9660 LUV_NETBOOT_MODULES
 
 build-grub-%: autogen-grub
 	$(Q)mkdir '$(workdir)/grub-build-$*'
@@ -125,7 +125,7 @@ build-grub-%: autogen-grub
 	$(call mk_faketarget)
 
 mkimage-grub-%: build-grub-%
-	$(Q)bits-grub-mkimage -O $* --output='$(target)/$($*-img)' --prefix=/boot/grub -d '$(workdir)/grub-build-$*/grub-core' $($*-extra-modules) $(common-modules)
+	$(Q)bits-grub-mkimage -O $* --output='$(target)/$($*-img)' --prefix="LUV_NETBOOT_ROOT/boot/grub" -d '$(workdir)/grub-build-$*/grub-core' $($*-extra-modules) $(common-modules)
 	$(call mk_faketarget)

 
-- 
1.9.1

